iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/MergedDanglingLineXml.java

```diff
@@ -0,0 +1,90 @@
+        adder.setId(id)
+                .setName(name);
+        boolean fictitious = XmlUtil.readOptionalBoolAttribute(context.getReader(), "fictitious", false);
+        adder.setFictitious(fictitious);
+        double p = XmlUtil.readOptionalDoubleAttribute(context.getReader(), "p");
+        double q = XmlUtil.readOptionalDoubleAttribute(context.getReader(), "q");
+        subElements.add(dl -> dl.getParentTerminal().ifPresent(t -> t.setP(p)));
+        subElements.add(dl -> dl.getParentTerminal().ifPresent(t -> t.setQ(q)));
+        readElement(id, adder, context);
+
+        readUntilEndRootElement(context.getReader(), () -> {
+            if (context.getReader().getLocalName().equals(PropertiesXml.PROPERTY)) {
+                String pName = context.getReader().getAttributeValue(null, PropertiesXml.NAME);
+                String value = context.getReader().getAttributeValue(null, PropertiesXml.VALUE);
+                subElements.add(dl -> dl.setProperty(pName, value));
+            } else if (context.getReader().getLocalName().equals(AliasesXml.ALIAS)) {
+                String aliasType = context.getReader().getAttributeValue(null, "type");
+                String alias = context.getAnonymizer().deanonymizeString(context.getReader().getElementText());
+                subElements.add(dl -> dl.addAlias(alias, aliasType));
+            } else {
+                throw new PowsyblException("Unknown element name <" + context.getReader().getLocalName() + "> in <" + id + ">");
+            }
+        });
+    }
+
+    @Override
+    protected String getRootElementName() {
+        return DanglingLineXml.ROOT_ELEMENT_NAME + side;
+    }
+
+    @Override
+    protected boolean hasSubElements(DanglingLine dl) {
+        throw new IllegalStateException("Should not be called");
+    }
+
+    @Override
+    protected boolean hasSubElements(DanglingLine identifiable, NetworkXmlWriterContext context) {
+        return hasValidGeneration(identifiable, context);
+    }
+
+    @Override
+    protected void writeRootElementAttributes(DanglingLine identifiable, TieLine parent, NetworkXmlWriterContext context) throws XMLStreamException {
+        DanglingLineXml.writeRootElementAttributesInternal(identifiable, () -> identifiable.getParentTerminal().orElseThrow(IllegalStateException::new), context);
+    }
+
+    @Override
+    protected MergedDanglingLineAdder createAdder(TieLine parent) {
+        throw new UnsupportedOperationException(); // is never called
+    }
+
+    @Override
+    protected DanglingLine readRootElementAttributes(MergedDanglingLineAdder adder, NetworkXmlReaderContext context) {
+        throw new UnsupportedOperationException(); // is never called
+    }
+
+    @Override
+    protected void readElement(String id, MergedDanglingLineAdder adder, NetworkXmlReaderContext context) {
+        DanglingLineXml.readRootElementAttributesInternal(adder, context);
```

Comment:
As the MergedDanglingLine is more conventional, couldn't we use the standard way with readRootElementAttributes and avoid overriding readElement?

Reply:
I think it is not possible, we are not able to implement a readRootElementAttributes that return a DanglingLine in this case (dangling line created from tielineAdder).